.gitignore

```diff
@@ -18,3 +18,5 @@ bundle.zip
 Vagrantfile
 .vagrant
 .idea
+
+vendor
```

Comment:
if this project is going to produce binaries, then why is `vendor` excluded?

Reply:
I've added a comment in the first commit, this also doesn't seem like a hard suggestion from dep (https://github.com/golang/dep/blob/master/docs/FAQ.md#should-i-commit-my-vendor-directory).

Reply:
some consequences of NOT committing a vendor directory are::
(1) reproducible builds are difficult to do offline if you don't already have vendored sources sitting somewhere local
(2) CI builds will have external dependencies because they need to download external sources for every build. this can also make CI builds slower
(3) if the project isn't pinning **exact** dependency versions via `dep` config, then dependencies can drift from build to build across different systems leading to subtle bugs (by default, dependencies via `dep` are not pinned to exact versions, there's an implicit `^` prefix injected when referencing project dependencies by semantic version)

Reply:
About dependencies drifting, we could specify tags, have a CI step which fails if the lock file gets altered after dep ensure, there should be solutions to prevent this.

This is just my personal preference, I'll make sure to raise your point in our team sync and we'll see what the consensus is.

Reply:
I second that committing the vendor directory solves and guards against a lot of problems.

Reply:
If by problems you mean for eg. a dependency remote repo getting deleted, committing dependencies would hide the problems instead of solving them imo. These kind of problems would only get stacked until someone eventually needs to bump dependencies and actually solve them (by using another dep, finding a fork, moving it to the project codebase, or other mean).

I'd rather see our nightly integration tests failing in such situations, so we can be notified early and take appropriate action.

We also have pip dependencies in the Python CLI, eg. docopt, requests, etc. The setup is not ideal right now as we don't require tagged versions, but would you also argue that we should commit them as part of our codebase?

@klueska @ArmandGrillet  wdyt?

Reply:
From security perspective its a problem if we cannot build new version of program if one of the dependencies isn't available because of repo deletion. There is a possibility of new `golang` version coming out with a security fix that would require us to rebuild binary in order to get security fix in place.

Committing `/vendor` directory is a standard practice in `golang` projects and most of ours projects (if not all) are following same pattern. We should be consistent in order to not confuse developers.

https://github.com/mesosphere/dcos-secrets
https://github.com/mesosphere/dcos-go-enterprise
https://github.com/mesosphere/dcos-cluster-linker
https://github.com/mesosphere/dcos-ca

Reply:
ok, I've added dependencies to this PR.